File: leetcode/1897. Redistribute Characters to Make All Strings Equal/1897.go
// https://helloacm.com/redistribute-characters-to-make-all-strings-equal/
// https://leetcode.com/problems/redistribute-characters-to-make-all-strings-equal/
// EASY, STRING, HASH MAP

func makeEqual(words []string) bool {
    var c = make(map[rune]int)
    for _, s := range words {
        for _, ch := range s {
            c[ch] ++
        }
    }
    for _, v := range c {
        if v % len(words) != 0 {
            return false
        }
    }
    return true
}

File: leetcode/1897. Redistribute Characters to Make All Strings Equal/1897.py
# https://helloacm.com/teaching-kids-programming-redistribute-characters-to-make-all-strings-equal/
# https://leetcode.com/problems/redistribute-characters-to-make-all-strings-equal/
# EASY, STRING, HASH MAP

class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        return all([x % len(words) == 0 for x in Counter(''.join(words)).values()])
